datadog/data_source_datadog_security_monitoring_rules.go

```diff
@@ -0,0 +1,246 @@
+package datadog
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"sort"
+	"strings"
+
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+)
+
+func dataSourceDatadogSecurityMonitoringRules() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDatadogSecurityMonitoringRulesRead,
+
+		Schema: map[string]*schema.Schema{
+			// Filters
+			"name_filter": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "A rule name to limit the search",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"tags_filter": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "A list of tags to limit the search",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"default_only_filter": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Limit the search to default rules",
+			},
+			"user_only_filter": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Limit the search to user rules",
+			},
+
+			// Computed
+			"rule_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"rules": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: datadogSecurityMonitoringRuleSchema(),
+				},
+			},
+		},
+	}
+}
+
+func dataSourceDatadogSecurityMonitoringRulesRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	var nameFilter *string
+	var defaultFilter *bool
+	var tagFilter map[string]bool
+
+	if v, ok := d.GetOk("name_filter"); ok {
+		filter := v.(string)
+		nameFilter = &filter
+	}
+
+	_, filterDefault := d.GetOk("default_only_filter")
+	_, filterUser := d.GetOk("user_only_filter")
+	if filterDefault && filterUser {
+		return errors.New("error: cannot filter both default and user rules")
```

Comment:
If you keep both attributes think you might be able to handle that with `ConflictsWith: []string{"user_only_filter"}` in `default_only_filter` schema definition.